### PR TITLE
Assert types before trying to convert data

### PR DESCRIPTION
**Pull request by gitnik**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| Doc updated   | no
| BC breaks?    | yes 
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/schmittjoh/serializer/issues/880
| License       | MIT

As suggested in https://github.com/schmittjoh/serializer/issues/880, I used the naive approach and just assert the type of all scalars when deserializing. Feedback welcome as I'm not sure if this is the best approach.

Also I'm not sure to what degree you would consider this to be a BC break, would love to have your input on that.

### CODE REVIEW

**Comment by goetas** [comment]
I see this as a recurring problem (see also #1173) and probably should try to come up with a backward compatible solution

File: composer.json [hunk 1]
```diff
@@ -17,6 +17,8 @@
     ],
     "require": {
         "php": "^7.2",
+        "ext-simplexml":" *",
```
Comment (goetas) [inline composer.json]: This dependencies are optional (if someone uses it only for json), thue should be removed

**Comment by goetas** [comment]
I'm willing to accept this PR even for the 3.x branch. Can you add tests please?

**Comment by c0nst4nt** [comment]
Any updates? Would be nice to have this improvement!

**Comment by blackandred** [comment]
Are there any updates? :wink:

**Comment by goetas** [comment]
Is anyone willing to continue the work on this and add tests?

**Comment by scaytrase** [comment]
@goetas I've created #1317 Feel free to tell me If we need more tests or new are located in wrong place

**Comment by goetas** [comment]
done in https://github.com/schmittjoh/serializer/pull/1317

**Pull request closed without merging**